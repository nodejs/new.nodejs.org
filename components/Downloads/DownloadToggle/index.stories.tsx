import { useState } from 'react';
import DownloadToggle from './index';
import type { FC } from 'react';
import type { DownloadToggleProps } from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

const Template: FC<DownloadToggleProps> = ({ selected, showDescription }) => {
  const [selectedTypeRelease, setSelectedTypeRelease] = useState(selected);

  const handleTypeReleaseToggle = (value: string) => {
    setSelectedTypeRelease(value);
  };

  return (
    <DownloadToggle
      selected={selectedTypeRelease}
      handleClick={handleTypeReleaseToggle}
      showDescription={showDescription}
    />
  );
};

type Story = StoryObj<typeof Template>;
type Meta = MetaObj<typeof Template>;

export const Default: Story = {
  args: { selected: 'LTS' },
};

export const Current: Story = {
  args: { selected: 'CURRENT' },
};

export default {
  component: Template,
  args: { showDescription: true },
} as Meta;
