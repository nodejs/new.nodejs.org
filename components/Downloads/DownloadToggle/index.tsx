import { FormattedMessage, useIntl } from 'react-intl';
import classnames from 'classnames';
import styles from './index.module.scss';
import type { FC } from 'react';

export type DownloadToggleProps = {
  handleClick: (type: string) => void;
  selected: string;
  showDescription?: boolean;
};

const DownloadToggle: FC<DownloadToggleProps> = ({
  handleClick,
  selected,
  showDescription = true,
}) => {
  const intl = useIntl();

  const activeClassNames = classnames({ [styles.active]: selected === 'LTS' });
  const currentClassNames = classnames(styles.current, {
    [styles.active]: selected === 'CURRENT',
  });

  const handleOnClick = () =>
    handleClick(selected === 'CURRENT' ? 'LTS' : 'CURRENT');

  return (
    <div className={styles.downloadToggle}>
      <div className={styles.selector}>
        <div className={styles.switch}>
          <button
            className={activeClassNames}
            type="button"
            role="switch"
            aria-label={intl.formatMessage({
              id: 'components.downloads.downloadToggle.ltsVersions',
            })}
            aria-checked={selected === 'LTS'}
            onClick={handleOnClick}
          >
            <FormattedMessage id="components.downloads.downloadToggle.lts" />
          </button>
          <button
            className={currentClassNames}
            type="button"
            role="switch"
            aria-label={intl.formatMessage({
              id: 'components.downloads.downloadToggle.currentVersions',
            })}
            aria-checked={selected === 'CURRENT'}
            onClick={handleOnClick}
          >
            <FormattedMessage id="components.downloads.downloadToggle.current" />
          </button>
        </div>
      </div>
      {showDescription && (
        <p className={styles.description}>
          <FormattedMessage
            id="components.downloads.downloadToggle.recommendation"
            values={{ selected }}
          />
        </p>
      )}
    </div>
  );
};

export default DownloadToggle;
