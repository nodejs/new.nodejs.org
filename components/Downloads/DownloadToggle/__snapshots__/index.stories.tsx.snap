// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloads/DownloadToggle Current smoke-test 1`] = `
<div class="DownloadToggle_downloadToggle__uZIub">
  <div class="DownloadToggle_selector__bskfX">
    <div class="DownloadToggle_switch__aIxmK">
      <button class
              type="button"
              role="switch"
              aria-label="LTS Versions"
              aria-checked="false"
      >
        LTS
      </button>
      <button class="DownloadToggle_current__vxN36 DownloadToggle_active__vW3Bp"
              type="button"
              role="switch"
              aria-label="Current Versions"
              aria-checked="true"
      >
        Current
      </button>
    </div>
  </div>
  <p class="DownloadToggle_description__M8tDG">
    With the latest features
  </p>
</div>
`;

exports[`Downloads/DownloadToggle Default smoke-test 1`] = `
<div class="DownloadToggle_downloadToggle__uZIub">
  <div class="DownloadToggle_selector__bskfX">
    <div class="DownloadToggle_switch__aIxmK">
      <button class="DownloadToggle_active__vW3Bp"
              type="button"
              role="switch"
              aria-label="LTS Versions"
              aria-checked="true"
      >
        LTS
      </button>
      <button class="DownloadToggle_current__vxN36"
              type="button"
              role="switch"
              aria-label="Current Versions"
              aria-checked="false"
      >
        Current
      </button>
    </div>
  </div>
  <p class="DownloadToggle_description__M8tDG">
    Recommended for most users
  </p>
</div>
`;
